internal/internal_task_handlers.go

```diff
@@ -621,24 +614,28 @@ func (wth *workflowTaskHandlerImpl) ProcessWorkflowTask(
 			zap.Int64("PreviousStartedEventId", task.GetPreviousStartedEventId()))
 	})
 
-	workflowContext, skipReplayCheck, err := wth.getOrCreateWorkflowContext(task, historyIterator)
+	workflowContext, err := wth.getOrCreateWorkflowContext(task, historyIterator)
 	if err != nil {
 		return nil, nil, err
 	}
-	defer workflowContext.release()
-
-	workflowTask := &workflowTask{task: task, historyIterator: historyIterator}
+	defer workflowContext.Unlock()
```

Comment:
Where is it locked in this method? Can code be refactored to make locking/unlocking pare visible?

Reply:
The lock is obtained in the getOrCreate() method because there are some operations that happen in that method that need to be done with lock.